### PR TITLE
chore: Added 2-step address lookup

### PR DESCRIPTION
**Pull request by goergisn**
## Summary
<changed>

For cards, the address lookup feature now supports a 2-step lookup process (for example, used by the Google Maps API): 

- First, it allows providing suggested addresses (that might be incomplete)
- Then, optionally, it allows making another API request to complete the suggestions.

You must change the callback as follows:

| 5.3.0 or earlier        | 5.4.0                          |
|-------------------------|--------------------------------|
| .lookup(LookupProvider) | .lookup(AddressLookupProvider) |

</changed>

### Changes
- Adds an optional callback that allows completion of an incomplete address (required for GoogleMaps autocompletion API)
- Moving the logic of address lookup search logic to its own ViewModel

### CODE REVIEW

File: spell-check-excluded-files-list [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Adyen/Utilities/External/AllRegions.swift
 Adyen/Core/Payment Methods/Abstract/PaymentMethodType.swift
 Adyen/Core/Models/CardType.swift
-Demo/Common/IntegrationExamples/AddressLookupProvider.swift
```
Comment (descorp) [inline spell-check-excluded-files-list:4]: Not sure if this is necessary
It is nice to know that we don't have typos in Demo code 🤔

File: Adyen/UI/View Controllers/AddressLookup/AddressLookupViewController.swift [hunk 2]
```diff
@@ -112,4 +95,19 @@ private extension AddressLookupViewController.ViewModel {
             completionHandler: handleAddressInputFormCompletion(validAddress:)
         )
     }
+    
+    func addressSearchViewModel(
```
Comment (descorp) [inline Adyen/UI/View Controllers/AddressLookup/AddressLookupViewController.swift]: Minor: function name supposed to start with verb  🤓

How about:
`getAddressSearchViewModel` or `buildAddressSearchViewModel` ?

Same applicable to `func addressInputFormViewModel`.

**Review by descorp — DISMISSED** [review]

File: Adyen/UI/View Controllers/AddressLookup/AddressLookupViewController.swift (hunk 2 above)
Comment (goergisn) [reply Adyen/UI/View Controllers/AddressLookup/AddressLookupViewController.swift]: Yes! Sounds good, will change 👌

Comment (goergisn) [reply Adyen/UI/View Controllers/AddressLookup/AddressLookupViewController.swift]: Fixed in c38ed109abaa17814809d01660c6cbf26f83ca47

File: Adyen/UI/View Controllers/AddressLookup/Search/AddressLookupSearchViewController+ViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+private extension AddressLookupSearchViewController.ViewModel {
+    
+    var manualEntryListItem: ListItem {
+        
+        .init(
+            title: localizedString(.addressLookupSearchManualEntryItemTitle, localizationParameters),
+            style: style.manualEntryListItem
+        ) {
+            self.switchToManualEntryHandler()
```
Comment (erenbesel) [inline Adyen/UI/View Controllers/AddressLookup/Search/AddressLookupSearchViewController+ViewModel.swift:117]: no need for weak self?

File: Adyen/UI/View Controllers/AddressLookup/Search/AddressLookupSearchViewController+ViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        let formattedLocation = address.formattedLocation(using: localizationParameters)
+        
+        let title = !formattedStreet.isEmpty ? formattedStreet : formattedLocation
+        let subtitle = !formattedStreet.isEmpty ? formattedLocation : nil
+        
+        let listItem = ListItem(title: title, subtitle: subtitle)
+        
+        listItem.selectionHandler = { [weak listItem] in
+            guard let listItem else { return }
+            self.handleDidSelect(item: listItem, addressModel: addressModel)
```
Comment (erenbesel) [inline Adyen/UI/View Controllers/AddressLookup/Search/AddressLookupSearchViewController+ViewModel.swift:137]: weak self here as well

File: Adyen/UI/View Controllers/AddressLookup/Search/AddressLookupSearchViewController+ViewModel.swift (hunk 3 above)
Comment (goergisn) [reply Adyen/UI/View Controllers/AddressLookup/Search/AddressLookupSearchViewController+ViewModel.swift:117]: Good catch! Not needed here as the ViewModel is a struct 👌

File: Adyen/UI/View Controllers/AddressLookup/Search/AddressLookupSearchViewController+ViewModel.swift (hunk 4 above)
Comment (goergisn) [reply Adyen/UI/View Controllers/AddressLookup/Search/AddressLookupSearchViewController+ViewModel.swift:137]: Same as above 😌

File: Adyen/UI/View Controllers/AddressLookup/Search/AddressLookupSearchViewController+ViewModel.swift (hunk 3 above)
Comment (erenbesel) [reply Adyen/UI/View Controllers/AddressLookup/Search/AddressLookupSearchViewController+ViewModel.swift:117]: ah nice ones then 👍

**Review by erenbesel — APPROVED** [review]

**Pull request merged**